shub/auth.py

```diff
@@ -0,0 +1,48 @@
+import os
+import netrc
+
+from os.path import expanduser
+
+from click import ClickException
+
+OS_WIN = True if os.name == 'nt' else False
+NETRC_FILE = expanduser('~/_netrc') if OS_WIN else expanduser('~/.netrc')
+
+
+def find_api_key():
+    """
+    Raises:
+        ClickException: if no credentials are found
+    """
+    key = os.getenv("SHUB_APIKEY")
```

Comment:
Yeah, it's fine to leave support for `SHUB_APIKEY` as long as the "logged in" API superseedes it.

The `SHUB_APIKEY` env var should also be used to populate the default API key for login.